checks/evaluation/branch_protection.go

```diff
@@ -411,19 +382,26 @@ func nonAdminThoroughReviewProtection(branch *clients.BranchRef, dl checker.Deta
 	log := branch.Protected != nil && *branch.Protected
 
 	max++
-	if branch.BranchProtectionRule.RequiredPullRequestReviews.RequiredApprovingReviewCount != nil {
-		switch *branch.BranchProtectionRule.RequiredPullRequestReviews.RequiredApprovingReviewCount >= minReviews {
-		case true:
+
+	// On this first check we exclude 1. The case which we don't know if the PRs are required
+	// 2. The case we know that PRs are not required, because it's covered on adminReviewProtection function.
+	if branch.BranchProtectionRule.RequiredPullRequestReviews != nil &&
+		branch.BranchProtectionRule.RequiredPullRequestReviews.RequiredApprovingReviewCount != nil {
+		switch {
+		// If not running as admin, the nil value can both mean that no reviews are required or no PR are required,
+		// so here we assume no reviews are required.
+		case *branch.BranchProtectionRule.RequiredPullRequestReviews.RequiredApprovingReviewCount == 0:
+			warn(dl, log, "number of required reviewers is 0 on branch '%s'", *branch.Name)
```

Comment:
You're right, we don't really need it.
I kept this specific case for 0 because on the default it would read as "number of required reviewers is only 0 on branch main" and it seemed a bit weird.
We can change the default to display like "number of required reviews is 0, while the ideal suggested is `minReviews`". WDYT?

Reply:
Solved on https://github.com/ossf/scorecard/pull/3499/commits/80f60e941995a1cad0daba214890f4bebd98e5e2